packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts

```diff
@@ -1,33 +1,232 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens, typographyStyles } from '@fluentui/react-theme';
 import type { PersonaSlots, PersonaState } from './Persona.types';
 import type { SlotClassNames } from '@fluentui/react-utilities';
 
 export const personaClassNames: SlotClassNames<PersonaSlots> = {
   root: 'fui-Persona',
-  // TODO: add class names for all slots on PersonaSlots.
-  // Should be of the form `<slotName>: 'fui-Persona__<slotName>`
+  avatar: 'fui-Persona-avatar',
+  presence: 'fui-Persona-presence',
+  primaryText: 'fui-Persona-primaryText',
+  secondaryText: 'fui-Persona-secondaryText',
+  tertiaryText: 'fui-Persona-tertiaryText',
+  quaternaryText: 'fui-Persona-quaternaryText',
 };
 
+export const COIN_GRID_AREA = 'coin';
+export const PRIMARY_TEXT_GRID_AREA = 'primary-text';
+export const SECONDARY_TEXT_GRID_AREA = 'secondary-text';
+export const TERTIARY_TEXT_GRID_AREA = 'tertiary-text';
+export const QUATERNARY_TEXT_GRID_AREA = 'quaternary-text';
+
 /**
  * Styles for the root slot
  */
 const useStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    display: 'inline-grid',
+    gridAutoColumns: 'max-content',
+    gridAutoRows: 'max-content',
+  },
+});
+
+const useGridStyles = makeStyles({
+  after: {
+    gridTemplateAreas: `
+    "${COIN_GRID_AREA} ${PRIMARY_TEXT_GRID_AREA}"
+    "${COIN_GRID_AREA} ${SECONDARY_TEXT_GRID_AREA}"
+    "${COIN_GRID_AREA} ${TERTIARY_TEXT_GRID_AREA}"
+    "${COIN_GRID_AREA} ${QUATERNARY_TEXT_GRID_AREA}"
+    `,
```

Comment:
Instead of using `gridTemplateAreas`, you can rely on grid's auto-flow behavior, and the DOM order.

* By default, root only needs `display: 'inline-grid'`.
* If `textPosition !== 'below'`
  * Set the root's `gridAutoFlow: 'column'` (the default grid flow of 'row' works for text below).
  * Set the avatar/presence slots' `gridRowStart: 'span 5'` (5 rows lets it extend one row past the text lines, so a large avatar doesn't mess up the height of the rows for the text lines)
* If `textPosition === 'before'`
  * When rendering, re-order the DOM to put the text slots before the avatar/presence badge slot.

See an example here: https://codepen.io/behowell/pen/RwyoXqz?editors=1100